piece-retriever/bin/piece-retriever.js

```diff
@@ -74,7 +74,7 @@ export default {
       const [{ serviceProviderId, serviceUrl, dataSetId }, isBadBit] =
         await Promise.all([
           getStorageProviderAndValidatePayer(env, payerWalletAddress, pieceCid),
-          env.BAD_BITS_KV.get(`bad-bits:${getBadBitsEntry(pieceCid)}`, {
+          env.BAD_BITS_KV.get(`bad-bits:${await getBadBitsEntry(pieceCid)}`, {
```

Comment:
Nitpick: this will await `getBadBitsEntry` first, and only then start `getStorageProviderAndValidatePayer` and `env.BAD_BITS_KV.get` in parallel.

IMO, a better solution would be to refactor `getBadBitsEntry` and `env.BAD_BITS_KV.get` into a standalone function, so that both legs can run concurrently from the beginning.

It makes me wonder if such a change would have a measurable impact on the retrieval response times 🤔 

See also `findInBadBits(env, cid)` we used to use before:
https://github.com/filbeam/worker/blob/e5681315b6d81d542d51966710cef3cca825a2d4/piece-retriever/lib/bad-bits-util.js#L19-L27